staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/validation/validation.go

```diff
@@ -77,6 +77,9 @@ func ValidateLabels(labels map[string]string, fldPath *field.Path) field.ErrorLi
 
 func ValidateDeleteOptions(options *metav1.DeleteOptions) field.ErrorList {
 	allErrs := field.ErrorList{}
+	if options.GracePeriodSeconds != nil && *options.GracePeriodSeconds < 0 {
```

Comment:
What happened when this came in invalid before?  We can really only tighten validation like this when the previous behavior was wrong/broken because otherwise it can can result in previously succeeding requests suddenly failing.

Reply:
@deads2k 
The previous behavior of the system before we tighten validation rules was `BeforeDelete` function which defined in `staging/src/k8s.io/apiserver/pkg/registry/rest/delete.go`. It called `ValidateDeleteOptions` at https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apiserver/pkg/registry/rest/delete.go#L73. On https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apiserver/pkg/registry/rest/delete.go#L103, it uses `GracePeriodSeconds` to do other checks without validating. If any invalid options were caught, `BeforeDelete` function would return an error.

Reply:
@huangjiuyuan but you haven't describe the client-observed behavior.  Was the value ignored, was it defaulted, did the apiserver panic, return an error, or do something else?